File: 프로그래머스/프로그래머스_네트워크.js
function bfs(graph, v, start) {
  v[start] = true;
  var Q = [start];

  while (Q.length) {
    const now = Q.shift();
    for (let i = 0; i < graph[now].length; i++) {
      if (!v[i] && graph[now][i]) {
        v[i] = true;
        Q.push(i);
      }
    }
  }
}

function solution(n, computers) {
  var visitied = new Array(n).fill(false);
  var answer = 0;

  for (let i = 0; i < n; i++) {
    if (!visitied[i]) {
      answer++;
      bfs(computers, visitied, i);
    }
  }

  return answer;
}

File: 그래프탐색/프로그래머스_네트워크.py
from collections import deque

def solution(n, computers):
    def BFS(k):
        nonlocal answer
        answer += 1
        Q = deque([k])
        while Q:
            nr = Q.popleft()

            for nc in range(0, n):

                if computers[nr][nc] and not v[nc]:
                    v[nc] = 1
                    Q.append(nc)

    v = [0] * n
    answer = 0

    for i in range(n):
        for j in range(n):
            if computers[i][j] and not v[j]:
                BFS(i)

    print(answer)
    return answer


n = 3
computers = [[1, 1, 0], [1, 1, 0], [0, 0, 1]]

solution(n,computers)